host_core/test/host_core/providers_test.exs

```diff
@@ -15,11 +15,13 @@ defmodule HostCore.ProvidersTest do
         @httpserver_contract
       )
 
+    Process.sleep(1000)
     assert HostCore.Providers.ProviderSupervisor.all_providers() == [
              {@httpserver_key, @httpserver_link, @httpserver_contract}
            ]
 
     HostCore.Providers.ProviderSupervisor.terminate_provider(@httpserver_key, @httpserver_link)
+    Process.sleep(500)
```

Comment:
I'm ok with this for the time being, but I think we should write up an issue to be able to monitor the event stream or otherwise `wait` for certain events to happen. I'm imagining us writing up some code in the future to start an actor and provider, and to think it may be a race condition to invoke that actor/provider immediately feels bad

Reply:
Wrote up https://github.com/wasmCloud/wasmcloud-otp/issues/29

Reply:
Agreed. Sleeps are a hack and a regression waiting to happen.